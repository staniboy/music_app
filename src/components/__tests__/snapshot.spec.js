import SongItem from "@/components/SongItem.vue";
import { RouterLinkStub, shallowMount } from "@vue/test-utils";
import { describe, expect, test } from "vitest";
describe("Snapshots SongItem.vue", () => {
  test("renders correctly", () => {
    const song = {
      id: "01234",
      modified_name: "modified_name_test",
      display_name: "display_name_test",
      comment_count: 69,
    };

    const component = shallowMount(SongItem, {
      propsData: {
        song,
      },
      global: {
        components: {
          "router-link": RouterLinkStub,
        },
      },
    });

    expect(component.element).toMatchSnapshot();
  });
});
