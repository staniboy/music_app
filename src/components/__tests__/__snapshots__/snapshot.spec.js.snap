// Vitest Snapshot v1

exports[`Snapshots SongItem.vue > renders correctly 1`] = `
<li
  class="flex justify-between items-center p-3 pl-6 cursor-pointer transition duration-300 hover:bg-gray-50"
  id="song-id-01234"
>
  <div>
    <router-link-stub
      class="font-bold block text-gray-600"
      custom="false"
      to="[object Object]"
    />
    <span
      class="display_name text-gray-500 text-sm"
    >
      display_name_test
    </span>
  </div>
  <div
    class="text-gray-600 text-lg"
  >
    <router-link-stub
      custom="true"
      to="[object Object]"
    />
  </div>
</li>
`;
